runtime/drivers/sql_store.go

```diff
@@ -0,0 +1,33 @@
+package drivers
+
+import (
+	"context"
+)
+
+type Field struct {
+	Name string
+	Type string
+}
+
+type Schema []Field
+
+// SQLStore is implemented by drivers capable of running sql queries and generating an iterator to consume results.
+// In future the results can be produced in other formats like arrow as well.
+// May be call it DataWarehouse to differentiate from OLAP or postgres?
+type SQLStore interface {
```

Comment:
Seeing `SQLStore` does seem confusing, not really sure what's a better alternative though